microovn/ovn/cmd/commands.go

```diff
@@ -200,18 +200,18 @@ func ControllerCtl(s *state.State, args ...string) (string, error) {
 	return stdout, err
 }
 
-// OvsdbClient is a wrapper function that executes 'ovsdb-client' command. It first ensures that the database
-// is connected and returns error if the database is not connected within <connectTimeout> seconds. Then it runs
-// "ovsdb-client" command with timeout of <resultTimeout> seconds.
-// Argument "args" should contain array of strings with subcommand and other arguments that will be passed directly
+// OvsdbClient is a wrapper function that executes 'ovsdb-client' command. It automatically adds arguments for timeout
+// (defined by resultTimeout), and arguments for SSL setup (crt, key, CA cert).
+// Argument "args" should contain array of strings with subcommand, and other arguments that will be passed directly
 // to the "ovsdb-client". Note that it is not necessary to pass "-t" argument, as the timeout is automatically included.
-func OvsdbClient(s *state.State, dbSpec *OvsdbSpec, connectTimeout int, resultTimeout int, args ...string) (string, error) {
-	err := WaitForDBState(s, dbSpec, OvsdbConnected, connectTimeout)
-	if err != nil {
-		return "", err
+func OvsdbClient(s *state.State, resultTimeout int, args ...string) (string, error) {
+	cert, key := paths.PkiClientCertFiles()
+	arguments := []string{
+		"-t", strconv.Itoa(resultTimeout),
+		"-c", cert,
```

Comment:
Is this really needed?

The `ovsdb-client` tool is capable of communicating with the cluster through the Unix socket on one of the nodes with a DB member in the cluster.

Reply:
This addition makes this wrapper usable also on nodes that don't run central services.

Reply:
That is not a good reason. The test code already defines lists with central nodes and chassis nodes, so you know where you can get the information. The test also defines a "probe container" which also has this information locally.